### DISCUSSION TITLE
I could use some assistance. Updating our codebase to use poetry on a large codebase with over 5,000 users and multiple internal packages.

### DISCUSSION THREAD

**Discussion by ughstudios** [python-poetry/poetry#7630] [Q&A]
I need some help here. I am trying to do a few things here with poetry.

There are multiple python packages developed internally using the default python packaging tooling. But what I want is to update the code to use poetry. 

However, I do not want the package resolution functionality. Is there a way to turn this off? If it's required, the workflow for how its setup is very poor. If I want to include another package in edit mode (--editable, -e../path/to/package) then that package needs a pyproject.toml file setup for peotry with all of its requirements. We already have a custom pypi.org server that controls and restricts the max versions for each third-party package we use. It's also vendorized and cached on our local clusters. 

Let me give an example:

I have 4 packages
a
b
c
d

a depends on b, c, and d. All 4 of these packages are local to our git repository and can be accessed locally by engineers/devs.  Now if a wants to add b as a --editable dependency, then b also needs to have a pyproject.toml file. Is there a way around this?

There are two ways I want to use poetry:
1. Installing packages in parallel to our "main" python interpreter
2. Creating venv's faster by using a better caching system than our internal setup (which fails sometimes) and installing packages to those venv's in parallel

But this is very difficult because poetry requires each pyproject.toml file to include a list of the packages (as well as their versions). I don't want to have versions in a specific pyproject.toml file, the master pypi server already handles this for me.

Is this something that is realistic with poetry? As of now, it seems like the current scope of what I want to use it for, and what it's made for will not work together in harmony. 

What I want to use:
1. The caching functionality for venv's
2. Parallel installing
3. If there are dependency issues, I would like to find those as well, but it should not be a blocker. There should be a way to "force" poetry to continue installing even if there is a dependency issue. We know this works because it works in our current setup. It would be nice if it would just send a warning message to stdout instead of being a blocker. 

Is this possible?

**Comment by radoering** [comment]
> However, I do not want the package resolution functionality. Is there a way to turn this off? 

Dependency resolution is a prerequisite for installing dependencies. It does not make sense to turn it off.

> If I want to include another package in edit mode (--editable, -e../path/to/package) then that package needs a pyproject.toml file setup for peotry with all of its requirements.

You can add valid packages, which are not managed by poetry, to your project.

> But this is very difficult because poetry requires each pyproject.toml file to include a list of the packages (as well as their versions). I don't want to have versions in a specific pyproject.toml file, the master pypi server already handles this for me.

If you don't care for versions you can specify the version as `"*"` (any version). If your sub packages are only used in your main project and you want to specify the dependencies of your sub packages only in the main project, you don't have to specify any dependencies in the pyproject.toml files of your sub packages. (Of course that doesn't make sense if your sub package will be used in other projects and you don't want to define the dependencies of your sub package in each of these.)

> There should be a way to "force" poetry to continue installing even if there is a dependency issue. [...] Is this possible?

No.

**Reply by ughstudios** [reply]
> You can add valid packages, which are not managed by poetry, to your project.
How?

**Reply by radoering** [reply]
[poetry add](https://python-poetry.org/docs/cli/#add)
